heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCache.java

```diff
@@ -0,0 +1,187 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.twitter.heron.common.config.SystemConfig;
+import com.twitter.heron.metricsmgr.MetricsSinksConfig;
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsFilter;
+
+/**
+ * compatible with tmaster interface
+ * see heron/tmaster/src/cpp/manager/tmetrics-collector.h
+ */
+public class MetricsCache {
+  public static final String METRICS_SINKS_TMASTER_SINK = "tmaster-sink";
+  public static final String METRICS_SINKS_TMASTER_METRICS = "tmaster-metrics-type";
+
+  // logger
+  private static final Logger LOG = Logger.getLogger(CacheCore.class.getName());
+
+  private CacheCore cache = null;
+  private MetricsFilter metricNameType = null;
+
+  /**
+   * constructor
+   *
+   * @param systemConfig heron config
+   * @param sinksConfig sink config
+   */
+  public MetricsCache(SystemConfig systemConfig, MetricsSinksConfig sinksConfig) {
+    // metadata
+    metricNameType = new MetricsFilter();
+    Map<String, Object> sinksTmaster = sinksConfig.getConfigForSink(METRICS_SINKS_TMASTER_SINK);
+    @SuppressWarnings("unchecked")
+    Map<String, String> metricsType =
+        (Map<String, String>) sinksTmaster.get(METRICS_SINKS_TMASTER_METRICS);
+    for (Map.Entry<String, String> e : metricsType.entrySet()) {
```

Comment:
Instead of iterating on Entries it's typically more readable to iterate on keys:
```
for (String metricName : metricsType.keySet() {
  metricNameType.setMetricToType(metricName, TranslateFromString(metricsType.get(metricName));
}
```